src/render/idyll-display/components/code-cell.js

```diff
@@ -0,0 +1,77 @@
+import * as React from 'react';
+
+/**
+ * Renders editable code cell that executes on shift + enter
+ * @props - onExecute, onBlur, markup
+ */
+export default class EditableCodeCell extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      editing: false
+    };
+
+    this._cellCodeRef = React.createRef();
+  }
+
+  toggleEdit = e => {
+    e.stopPropagation();
+    if (this.state.editing) {
+      this.setState({
+        editing: false
+      });
+    } else {
+      this.edit();
+    }
+  };
+
+  edit = () => {
+    this.setState(
+      {
+        editing: true
+      },
+      () => {
+        this._cellCodeRef.current.focus();
+      }
+    );
+  };
+
+  // Executes code change
+  execute = () => {
+    this.props.onExecute(this._cellCodeRef.current.textContent);
+  };
+
+  // Updates markup on blur
+  onBlur = e => {
+    this.toggleEdit(e);
+    this.props.onBlur(this._cellCodeRef.current.textContent);
+  };
+
+  handleKeyDown = e => {
+    e.stopPropagation();
+    if (e.shiftKey && e.keyCode === 13) {
```

Comment:
We could also add these to a constants file (keycode constants)

Reply:
👍  added